pkg/scheduler/algorithm/priorities/requested_to_capacity_ratio.go

```diff
@@ -84,35 +86,64 @@ func NewFunctionShape(points []FunctionShapePoint) (FunctionShape, error) {
+			return nil, fmt.Errorf("resource %s weight %d must not be less than 0", string(resource), weight)
+		}
+	}
+	return resources, nil
+}
+
 // RequestedToCapacityRatioResourceAllocationPriorityDefault creates a requestedToCapacity based
 // ResourceAllocationPriority using default resource scoring function shape.
 // The default function assigns 1.0 to resource when all capacity is available
 // and 0.0 when requested amount is equal to capacity.
 func RequestedToCapacityRatioResourceAllocationPriorityDefault() *ResourceAllocationPriority {
-	return RequestedToCapacityRatioResourceAllocationPriority(defaultFunctionShape)
+	return RequestedToCapacityRatioResourceAllocationPriority(defaultFunctionShape, defaultResources)
 }
 
 // RequestedToCapacityRatioResourceAllocationPriority creates a requestedToCapacity based
 // ResourceAllocationPriority using provided resource scoring function shape.
-func RequestedToCapacityRatioResourceAllocationPriority(scoringFunctionShape FunctionShape) *ResourceAllocationPriority {
-	return &ResourceAllocationPriority{"RequestedToCapacityRatioResourceAllocationPriority", buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape)}
+func RequestedToCapacityRatioResourceAllocationPriority(scoringFunctionShape FunctionShape, resources Resources) *ResourceAllocationPriority {
+	return &ResourceAllocationPriority{"RequestedToCapacityRatioResourceAllocationPriority", buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape, resources), resources}
 }
 
-func buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape FunctionShape) func(*schedulernodeinfo.Resource, *schedulernodeinfo.Resource, bool, int, int) int64 {
+func buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape FunctionShape, resources Resources) func(map[v1.ResourceName]int64, map[v1.ResourceName]int64, bool, int, int) int64 {
 	rawScoringFunction := buildBrokenLinearFunction(scoringFunctionShape)
 
 	resourceScoringFunction := func(requested, capacity int64) int64 {
 		if capacity == 0 || requested > capacity {
 			return rawScoringFunction(maxUtilization)
 		}
 
+		//fmt.Printf("resourceAllocation %d\n", maxUtilization-(capacity-requested)*maxUtilization/capacity)
 		return rawScoringFunction(maxUtilization - (capacity-requested)*maxUtilization/capacity)
 	}
 
-	return func(requested, allocable *schedulernodeinfo.Resource, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
-		cpuScore := resourceScoringFunction(requested.MilliCPU, allocable.MilliCPU)
-		memoryScore := resourceScoringFunction(requested.Memory, allocable.Memory)
-		return (cpuScore + memoryScore) / 2
+	return func(requested, allocable map[v1.ResourceName]int64, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
+		var nodeScore, weightSum int64
+		nodeScore = 0
+		weightSum = 0
+		for resource, weight := range resources {
+			resourceScore := resourceScoringFunction(requested[resource], allocable[resource])
```

Comment:
does requested include the sum of all pods on the node, including the pod currently scheduled plus the pod being tried on this node ?

Reply:
yes